lib/puppet/provider/archive/curl.rb

```diff
@@ -25,7 +44,12 @@ def download(filepath)
       ]
     )
 
-    curl(params)
+    begin
+      curl(params)
+    rescue StandardError => e
```

Comment:
You can also use `ensure` which is more suited for this particular case.

Reply:
and more to the point, I've completely forgotten to call delete_netrcfile when the command didn't fail!!  Doh!

Reply:
Fixed.